### DISCUSSION TITLE
Tailwind override position values differently / inconsistently

### DISCUSSION THREAD

**Discussion by chrisj-skinner** [tailwindlabs/tailwindcss#12317] [Help] [answered]
I'm seeing some odd behaviour, inconsistency and unable to track the source down.

- No changes or additions to tailwind config theme.
- version `3.1.8`
- Using a positive and negative top position. Sometimes the positive will override the negative and sometime the reverse.

**First instance** with a value of `-256px` works as expected with `-70px` overriding when toggled.

**Second instance** with a value of `-258px`. `-70px` does not override when toggled.\

The DOM displays the same class names in both instances. It's very inconsistent, changing `-256px` but a couple of digits up or down works or doesn't.

Toggling the className via useState and an onClick
```
const [isActive, setState] = useState(true);
...
<button onClick={() => setState(!isActive)}>Toggle</button>
```

### First instance -256px
JSX ```<p className={`absolute w-60 top-[-256px] ${isActive ? 'top-[70px]' : '' }`} >``` 

isActive `false`
DOM `<p className="absolute w-60 top-[-256px]">` **As Expected**
DOM Styles  **As Expected**
![Screenshot 2023-10-29 at 14 02 55](https://github.com/tailwindlabs/tailwindcss/assets/5794048/e2754f43-6bbe-4314-9cb1-762329086854)

isActive `true`
DOM `<p className="absolute w-60 top-[-256px] top-[70px]">` **As Expected**
DOM Styles  **As Expected**
![Screenshot 2023-10-29 at 13 54 34](https://github.com/tailwindlabs/tailwindcss/assets/5794048/79ae3592-b460-4e96-8a63-4211f2616185)

### Second instance -258px
JSX ```<p className={`absolute w-60 top-[-258px] ${isActive ? 'top-[70px]' : '' }`} >```

isActive `false`
DOM `<p className="absolute w-60 top-[-258px]]">` **As Expected**
DOM Styles  **As Expected**
 ![Screenshot 2023-10-29 at 14 03 16](https://github.com/tailwindlabs/tailwindcss/assets/5794048/f5436904-57df-4ebf-96e0-71d84a0636cd)

isActive `true`
DOM `<p className="absolute w-60 top-[-258px] top-[70px]">` **As Expected**

DOM Styles **Not as Expected**
![Screenshot 2023-10-29 at 13 54 18](https://github.com/tailwindlabs/tailwindcss/assets/5794048/105b0264-fc4f-4807-b827-db49c0860ecb)

In this case `-70px` does not override `-258px`. Thoughts>?

**Comment by wongjn** [comment, marked as answer]
Consider avoiding having multiple class names that apply the same property with the same variant:

```jsx
<p className={`absolute w-60 ${isActive ? 'top-[70px]' : 'top-[-256px]' }`} >
```

This will mean only one `top-*` class will be applied at any one point, thus avoiding any CSS rule precedence jumbles.
